src/shared/tpm2-util.c

```diff
@@ -695,15 +673,85 @@ static int tpm2_get_policy_digest(
         return 0;
 }
 
+static int tpm2_pcr_read(
+                struct tpm2_context *c,
+                const TPML_PCR_SELECTION *pcr_selection,
+                TPML_PCR_SELECTION *ret_pcr_selection,
+                TPM2B_DIGEST **ret_pcr_values,
+                size_t *ret_pcr_values_size) {
+
+        _cleanup_free_ TPM2B_DIGEST *pcr_values = NULL;
+        TPML_PCR_SELECTION remaining, read = {};
```

Comment:
hmm, "read()" is a libc symbol. maybe pick a different name. old gcc used to complain about this, new doesn't anymore, but I#d still avoid it

Reply:
ack, changed it to 'total_read'